src/resdk/collection_tables.py

```diff
@@ -0,0 +1,274 @@
+""".. Ignore pydocstyle D400.
+
+=================
+Collection Tables
+=================
+
+.. autoclass:: CollectionTables
+    :members:
+
+    .. automethod:: __init__
+"""
+import os
+from datetime import datetime
+from functools import lru_cache
+from io import BytesIO
+from typing import Callable, Dict, List, NamedTuple, Optional
+from urllib.parse import urljoin
+
+import pandas as pd
+import pytz
+
+from resdk import Resolwe
+from resdk.resources import Collection
+from resdk.utils.table_cache import default_cache_dir, load_cache, save_cache
+
+TPM = "tpm"
+COUNTS = "rc"
+EXPRESSION_TYPES = [TPM, COUNTS]
+
+META = "meta"
+
+CHUNK_SIZE = 1000
+
+
+class _TableSample(NamedTuple):
+    sample_slug: str
+    expression_files: Dict[str, str]
+    metadata: dict
+
+
+class CollectionTables:
+    """A helper class to fetch given collection's expression and meta data in a tables format.
+
+    All returned tables have samples in rows and gene symbols in columns.
+
+    When calling :meth:`CollectionTables.expressions`, :meth:`CollectionTables.counts` and
+    :meth:`CollectionTables.metadata` for the first time the corresponding data gets fetched from the server.
+    This data than gets cached in memory and on disc and is used in consequent calls until the modified
+    attribute changes.
+    """
+
+    def __init__(self, resolwe_obj: Resolwe, collection_slug: str, cache_dir=None):
+        """Initialize class.
+
+        :param resolwe_obj: initialized Resolwe object
+        :param collection_slug: slug of collection of interest
+        :param cache_dir: cache directory location, if not specified system specific cache directory is used
+        """
+        self.resolwe_obj = resolwe_obj
```

Comment:
For consistency, I would prefer if this would be named just `resolwe`, as in all of the Resource classes.